lib/ui5Framework/Sapui5MavenSnapshotResolver.js

```diff
@@ -0,0 +1,207 @@
+	 * @param {boolean} [options.sources=false] Whether to install framework libraries as sources or
+	 * pre-built (with build manifest)
+	 * @param {string} [options.cwd=process.cwd()] Current working directory
+	 * @param {string} [options.ui5HomeDir="~/.ui5"] UI5 home directory location. This will be used to store packages,
+	 * metadata and configuration used by the resolvers. Relative to `process.cwd()`
+	 * @param {string} [options.cacheMode="default"] Can be "default" (cache everything, invalidate after 9 hours),
+	 * 	"off" (do not cache) and "force" (use cache only - no requests)
+	 * @param {string} [options.artifactsDir] Where to install Maven artifacts
+	 * @param {string} [options.packagesDir] Where to install packages
+	 * @param {string} [options.metadataDir] Where to store Maven's artifacts metadata
+	 * @param {string} [options.stagingDir] The staging directory for artifacts and packages
+	 */
+	constructor(options) {
+		super(options);
+
+		const snapshotEndpointUrl =
+			options.snapshotEndpointUrl ||
+			process.env.UI5_MAVEN_SNAPSHOT_ENDPOINT;
+		const {
+			cacheMode,
+			artifactsDir,
+			packagesDir,
+			metadataDir,
+			stagingDir,
+			sources,
+		} = options;
+
+		if (!snapshotEndpointUrl) {
+			this._snapshotEndpointUrlCb = Sapui5MavenSnapshotResolver._resolveSnapshotEndpointUrl;
+			// TODO: Save resolved snapshot URL when .ui5rc Configuration is implemented
+		} else {
+			this._snapshotEndpointUrlCb = () => Promise.resolve(snapshotEndpointUrl);
+		}
+
+		this._installer = new Installer({
+			ui5HomeDir: this._ui5HomeDir,
+			snapshotEndpointUrlCb: this._snapshotEndpointUrlCb,
+			cacheMode,
+			artifactsDir,
+			packagesDir,
+			metadataDir,
+			stagingDir,
+		});
+		this._loadDistMetadata = null;
+		this._sources = !!sources;
+	}
+	loadDistMetadata() {
+		if (!this._loadDistMetadata) {
+			this._loadDistMetadata = Promise.resolve().then(async () => {
+				const version = this._version;
+				log.verbose(
+					`Installing ${DIST_ARTIFACT_ID} in version ${version}...`
+				);
+
+				const {pkgPath: distPkgPath} = await this._installer.installPackage({
+					pkgName: DIST_PKG_NAME,
+					groupId: DIST_GROUP_ID,
+					artifactId: DIST_ARTIFACT_ID,
+					version,
+					classifier: "npm-sources",
+					extension: "zip",
+				});
+
+				return await this._installer.readJson(
+					path.join(distPkgPath, "metadata.json")
+				);
+			});
+		}
+		return this._loadDistMetadata;
+	}
+	async getLibraryMetadata(libraryName) {
+		const distMetadata = await this.loadDistMetadata();
+		const metadata = distMetadata.libraries[libraryName];
+
+		if (!metadata) {
+			throw new Error(`Could not find library "${libraryName}"`);
+		}
+
+		return metadata;
+	}
+	async handleLibrary(libraryName) {
+		const metadata = await this.getLibraryMetadata(libraryName);
+		if (!metadata.gav) {
+			throw new Error(
+				`Metadata is missing GAV information. This might indicate an unsupported SNAPSHOT version.`
```

Comment:
Do people know what GAV stands for?

Reply:
I believe the ones that use this resolver should have that knowledge. However, now I'm wondering whether we could somehow give some context or even a link, for example: https://maven.apache.org/guides/mini/guide-naming-conventions.html

Reply:
Now it should be better